### DISCUSSION TITLE
Annotations

### DISCUSSION THREAD

**Discussion by andreachirico79** [burningwave/core#37] [Q&A]
HI, how can i implement the following annotation?

`@JsonInclude(JsonInclude.Include.NON_EMPTY)`

thanks

**Comment by Roberto-Gentili** [comment]
> JsonInclude.Include.NON_EMPTY
```java
AnnotationSourceGenerator.create(JsonInclude.class).addParameter(VariableSourceGenerator.create("JsonInclude.Include.NON_EMPTY"))
```

Check [here](https://github.com/burningwave/core/blob/master/src/test/java/org/burningwave/core/UnitSourceGeneratorTest.java) for complete examples about [source generators](https://github.com/burningwave/core/wiki/How-to-compile-sources-at-runtime)
